modules/kafka/src/main/scala/higherkindness/mu/format/Decoder.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.format
+
+trait Decoder[A] {
+  def decode(a: Array[Byte]): A
```

Comment:
I would expect this to return something like `Either[DecodeError, A]`

Reply:
Yes I agree. This PR is missing an error handling code. We will need to add appropriate actions when decoding fails. My suggestion is that we log it and throw away the message, ie commit the offset. All other approaches I have seen such as dead message topic, brings up its own complications. Or we can leave it to the end user.